nf_core/sync.py

```diff
@@ -333,53 +318,128 @@ def update_existing_pull_request(self, pr_title, pr_body_text):
+            log.info(
+                "Branch already existed: '{}', creating branch '{}' instead.".format(
+                    original_merge_branch, self.merge_branch
+                )
+            )
+
+        # Create new branch and checkout
+        log.info("Checking out merge base branch {}".format(self.merge_branch))
+        try:
+            self.repo.create_head(self.merge_branch)
+        except git.exc.GitCommandError as e:
+            raise SyncException(f"Could not create new branch '{self.merge_branch}'\n{e}")
+
+    def push_merge_branch(self):
+        """Push the newly created merge branch to the remote repository"""
+        log.info("Pushing {} branch to remote".format(self.merge_branch))
+        try:
+            origin = self.repo.remote()
+            origin.push(self.merge_branch)
+        except git.exc.GitCommandError as e:
+            raise PullRequestException(f"Could not push branch '{self.merge_branch}':\n  {e}")
+
+    def make_pull_request(self):
+        """Create a pull request to a base branch (default: dev),
+        from a head branch (default: TEMPLATE)
+
+        Returns: An instance of class requests.Response
+        """
+        # Check that we know the github username and repo name
+        if self.gh_username is None and self.gh_repo is None:
+            raise PullRequestException("Could not find GitHub username and repo name")
+
+        # If we've been asked to make a PR, check that we have the credentials
+        if os.environ.get("GITHUB_AUTH_TOKEN", "") == "":
+            raise PullRequestException(
+                "Environment variable GITHUB_AUTH_TOKEN not set - cannot make PR\n"
+                "Make a PR at the following URL:\n  https://github.com/{}/compare/{}...TEMPLATE".format(
+                    self.gh_repo, self.original_branch
+                )
+            )
+
+        log.info("Submitting a pull request via the GitHub API")
+
+        pr_title = "Important! Template update for nf-core/tools v{}".format(nf_core.__version__)
+        pr_body_text = (
+            "A new release of the main template in nf-core/tools has just been released. "
+            "This automated pull-request attempts to apply the relevant updates to this pipeline.\n\n"
+            "Please make sure to merge this pull-request as soon as possible. "
+            "Once complete, make a new minor release of your pipeline. "
+            "For instructions on how to merge this PR, please see "
+            "[https://nf-co.re/developers/sync](https://nf-co.re/developers/sync#merging-automated-prs).\n\n"
+            "For more information about this release of [nf-core/tools](https://github.com/nf-core/tools), "
+            "please see the [nf-core/tools v{tag} release page](https://github.com/nf-core/tools/releases/tag/{tag})."
+        ).format(tag=nf_core.__version__)
+
+        # Make new pull-request
+        self.submit_pull_request(pr_title, pr_body_text)
```

Comment:
I think that this can probably all be in the same function now - we only call this function once right? And especially if we remove the checks above this function will only define two strings 😅